src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Emitter.cs

```diff
@@ -65,29 +65,57 @@ private static bool UseLoggerMessageDefine(LoggerMethod lm)
 
             private void GenType(LoggerClass lc)
             {
+                string nestedIndentation = "";
                 if (!string.IsNullOrWhiteSpace(lc.Namespace))
                 {
                     _builder.Append($@"
 namespace {lc.Namespace}
 {{");
                 }
 
+                LoggerClass parent = lc.ParentClass;
+                var parentClasses = new List<string>();
+                // loop until you find top level nested class
+                while (parent != null)
+                {
+                    parentClasses.Add($"partial {parent?.Keyword} {parent?.Name} {parent?.Constraints}");
```

Comment:
Why do you need all these `?` on `parent`? You know it is non-null (from the loop condition), so you should be able to remove them.